test/email_test.py

```diff
@@ -29,3 +33,101 @@ def testEmailNoPrefix(self):
     @with_config({"core": {"email-prefix": "[prefix]"}})
     def testEmailPrefix(self):
         self.assertEqual("[prefix] subject", notifications._prefix('subject'))
+
+
+class TestException(Exception):
+    pass
+
+
+class TestTask(luigi.Task):
+    foo = luigi.Parameter()
+    bar = luigi.Parameter()
+
+
+class FailSchedulingTask(TestTask):
+    def requires(self):
+        raise TestException('Oops!')
+
+    def run(self):
+        pass
+
+    def complete(self):
+        return False
+
+
+class FailRunTask(TestTask):
+    def run(self):
+        raise TestException('Oops!')
+
+    def complete(self):
+        return False
+
+
+class ExceptionFormatTest(unittest.TestCase):
+
+    def setUp(self):
+        self.sch = CentralPlannerScheduler()
+        self.w = Worker(scheduler=self.sch)
+
+    def tear_down(self):
+        self.w.stop()
+
+    def test_fail_run(self):
+        task = FailRunTask(foo='foo', bar='bar')
+        self._run_task(task)
+
+    def test_fail_run_html(self):
+        task = FailRunTask(foo='foo', bar='bar')
+        self._run_task_html(task)
+
+    def test_fail_schedule(self):
+        task = FailSchedulingTask(foo='foo', bar='bar')
+        self._run_task(task)
+
+    def test_fail_schedule_html(self):
+        task = FailSchedulingTask(foo='foo', bar='bar')
+        self._run_task_html(task)
+
+    @with_config({'core': {'error-email': 'nowhere@example.com',
+                           'error-prefix': '[TEST] '}})
+    @mock.patch('luigi.notifications.send_error_email')
+    def _run_task(self, task, mock_send):
+        self.w.add(task)
+        self.w.run()
+
+        self.assertEqual(mock_send.call_count, 1)
+        args, kwargs = mock_send.call_args
+        self._check_subject(args[0], task)
+        self._check_body(args[1], task, html=False)
+
+    @with_config({'core': {'error-email': 'nowhere@axample.com',
```

Comment:
I'm fine with continuing to use `[core]` for this email. But where is error-prefix defined? And would you consider moving these stuff to a `[notifications]` section in a follow up PR? :)